src/mono/mono/mini/debugger-agent.c

```diff
@@ -6358,10 +5910,12 @@ do_invoke_method (DebuggerTlsData *tls, Buffer *buf, InvokeData *invoke, guint8
 		return ERR_INVALID_ARGUMENT;
 	} else if (m_class_is_valuetype (m->klass) && (m->flags & METHOD_ATTRIBUTE_STATIC)) {
 		/* Should be null */
-		int type = decode_byte (p, &p, end);
-		if (type != VALUE_TYPE_ID_NULL) {
-			PRINT_DEBUG_MSG (1, "[%p] Error: Static vtype method invoked with this argument.\n", (gpointer) (gsize) mono_native_thread_id_get ());
-			return ERR_INVALID_ARGUMENT;
+		if (!CHECK_PROTOCOL_VERSION (3, 0)) {
```

Comment:
I don't understand what's going on with static methods and protocol 3.0.  could you add a comment?

Reply:
yes